Remove typescriptServices.{js,d.ts}, protocol.d.ts, typescript_standalone.d.ts

jakebailey opened the pull request:
Fixes #50758, for TS 5.0.

These are files that nobody appears to use anymore, or are duplicated somewhere else. Notably, I have made all of these changes already on the module-ified TS branch, so this is a backport in prep for that (adapted for our current gulp build).

I'll try a `pack this` once https://github.com/microsoft/TypeScript-Make-Monaco-Builds/pull/14 is in.

TODO:

- [ ] Add to [microsoft/TypeScript/wiki/API-Breaking-Changes](https://github.com/microsoft/TypeScript/wiki/API-Breaking-Changes).

jakebailey commented:
@typescript-bot pack this

sheetalkamat commented:
protocol.d.ts: We created this for vs and vscode as far as I remember.. are they not using it.

jakebailey commented:
Hm, so our current LKG script appears to be unable to _delete_ files that are no longer copied, so this isn't yet a perfect test.

jakebailey commented:
Hah, so, this explains why Deno has in their "updating typescript" instructions a step that is deleting "dead" d.ts files. Our script never actually removed files that are no longer part of the build.

jakebailey commented:
@typescript-bot pack this

jakebailey commented:
Alright, seems to work now.

jakebailey commented:
@typescript-bot pack this

jakebailey commented:
My latest PR clears the lib folder (besides README) before copying into it, so that the files get deleted.

@weswigham mentioned that this might be undesirable; I can imagine that we actually do want to keep "dead" files forever, or something, so that people can continue to reference them in their old projects, but I'm not sure if that's intentional or not. If we do want this, it seems like we'd actually want to have those files in `src/lib` and declare that they really do exist.

For now, I can instead just change `produceLKG` to warn on files that don't appear to have come from any input, and then explicitly have it delete the files that this PR deletes.

jakebailey commented:
Looking at the diff when I run LKG locally, these files are deleted:

```
        deleted:    lib/lib.esnext.promise.d.ts
        deleted:    lib/lib.esnext.string.d.ts
        deleted:    lib/lib.esnext.weakref.d.ts
```

It appears as though we handle these explicitly:

https://github.com/microsoft/TypeScript/blob/5cd49f6cbcd2effe9d425dee3a39cb49209bb656/src/compiler/commandLineParser.ts#L94-L96

jakebailey commented:
For reference, the new package contents: https://unpkg.com/browse/@typescript-deploys/pr-build@4.9.0-pr-51026-13/lib/

jakebailey commented:
Pulled out the part of this that can be merged early into #51061.

jakebailey commented on scripts/createPlaygroundBuild.mjs:
```diff
@@ -6,7 +6,7 @@
 // This script does two things: 
 //  
-//  - Listens to changes to the built version of TypeScript (via a filewatcher on `built/local/typescriptServices.js`)
+//  - Listens to changes to the built version of TypeScript (via a filewatcher on `built/local/typescript.js`)
```
This will have to be updated to match the final state of monaco-editor. e.g. https://github.com/microsoft/monaco-editor/pull/3356

jakebailey commented:
@typescript-bot pack this

jakebailey commented:
> protocol.d.ts: We created this for vs and vscode as far as I remember.. are they not using it.

@sheetalkamat I totally misread your comment, I didn't realize you were asking if they were using this file.

It turns out that they are using it, though indirectly: https://github.com/microsoft/vscode/blob/main/extensions/typescript-language-features/src/protocol.d.ts

I'll have to see if we can instead have them write something like:

```ts
import * as ts from "typescript"
export = ts.server.protocol
```

Which I think should do the same thing.

jakebailey commented:
I sent https://github.com/microsoft/vscode/pull/163365 with this approach, which actually revealed a couple oops-es in the VS Code codebase.

jakebailey commented:
Going to close this; I will include this in the module conversion PR when the time comes.
